source/org/zfin/marker/presentation/SequenceTargetingReagentAddController.java

```diff
@@ -236,8 +236,10 @@ String proposeNameByTypeAndGenes(@RequestParam("type") String type, @RequestPara
         Set<String> geneSet = Set.of(genes.split(","));
         List<Marker> strList = mr.getMarkerWithRelationshipsBySecondMarkers(geneSet);
 
+        Marker.Type strType = Marker.Type.getTypeByPrefix(type);
+
         //filtered by type
-        strList = strList.stream().filter(str -> str.getMarkerType().getName().equals(type)).toList();
+        strList = strList.stream().filter(str -> str.getMarkerType().getName().equals(strType.toString())).collect(Collectors.toList());
```

Comment:
I think you can use a short cut when collecting into a List:.
collect(Collectors.toList())
can be written as
.toList()